### PR TITLE
PHOENIX-6531 Document Phoenixdb release process

### PR DESCRIPTION
**Pull request by stoty**
also remove sha256 signature generation

### CODE REVIEW

File: python-phoenixdb/RELEASING.rst [hunk 1]
```diff
@@ -1,12 +1,72 @@
...
+Make sure that your git remote ``origin`` points to either the main gitbox or the main github 
+phoenix-queryserver repo.
+
+Make sure that you have a PyPI account, and that you can publish to the 
+https://pypi.org/project/phoenixdb/ project. If not, then reach out to one of the maintainers listed there for permission.
+
+For instructions on the PyPi registration and publishing process, see
+https://kynan.github.io/blog/2020/05/23/how-to-upload-your-package-to-the-python-package-index-pypi-test-server
+
+Prepare the RC
```
Comment (virajjasani) [inline python-phoenixdb/RELEASING.rst:23]: Nice steps, very well explained!

**Review by virajjasani — APPROVED** [review]
If make_rc change is intentional, +1

**Comment by stoty** [comment]
Thanks @virajjasani 
The make_rc change is what you requested when voting for 1.0.1.

**Comment by virajjasani** [comment]
Oh right, now I recall this correctly. What a coincidence :)

**Pull request closed without merging**